src/video/wayland/SDL_waylandevents.c

```diff
@@ -883,7 +962,7 @@ Wayland_display_add_input(SDL_VideoData *d, uint32_t id)
         return;
 
     input->display = d;
-    input->seat = wl_registry_bind(d->registry, id, &wl_seat_interface, 1);
+    input->seat = wl_registry_bind(d->registry, id, &wl_seat_interface, 5);
```

Comment:
@bryphe this breaks backwards compat with some really old wayland compositor versions (not sure if present on some LTS distros) but they're old enough that they're likely not usable over their X based counterparts. Should switching logic be put in here before any merge happens?

Reply:
@szbergeron - I'm unfortunately not an expert on this!

What would the experience be if you had a really old wayland compositor? A crash? And how old are we talking / how likely is it to be hit?

I'd like to make sure it's diagnosable, at least on my end, in the case that we do hit this - perhaps detection + an SDL_Log?

Reply:
Hitting this makes the compositor kill the app and print a message saying that wl_seat versioning is incompatible to the console (reproducible by putting, say, a "6" in that position on a newer compositor)

The interface was added to the spec in late 2015 of this versioning (in commit https://github.com/wayland-project/wayland/commit/c5356e9016aa814a873a765bb2cbe57e804e5ea7) so any server newer than mid-2016 or so should at least support requesting that version, if not fully support it. Until middle of this year KDE didn't send out axis_stop events but would still provide for seats v5 requests. The callbacks in question just wouldn't get called :woman_shrugging:

Reply:
Thanks for the details! If we get a clear error message, I'm OK with it.